tests/vmi_hook_sidecar_test.go

```diff
@@ -228,6 +234,123 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
 	})
 })
 
+var _ = Describe("[sig-compute]HookQEMUArgsSidecar", decorators.SigCompute, func() {
+
+	var err error
+	var virtClient kubecli.KubevirtClient
+
+	var vmi *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+
+		vmi = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.ObjectMeta.Annotations = RenderValidQEMUArgsSidecar(hooksv1alpha1.Version)
+	})
+
+	Describe("[vendor:cnv-qe@redhat.com][level:component] VMI definition", func() {
+		getVMIPod := func(vmi *v1.VirtualMachineInstance) (*k8sv1.Pod, bool, error) {
+			podSelector := tests.UnfinishedVMIPodSelector(vmi)
+			vmiPods, err := virtClient.CoreV1().Pods(vmi.GetNamespace()).List(context.Background(), podSelector)
+
+			if err != nil {
+				return nil, false, fmt.Errorf("could not retrieve the VMI pod: %v", err)
+			} else if len(vmiPods.Items) == 0 {
+				return nil, false, nil
+			}
+			return &vmiPods.Items[0], true, nil
+		}
+
+		Context("with QEMU Args hook sidecar", func() {
+			It("should successfully start with hook sidecar annotation", func() {
+				By("Starting a VMI")
+				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				Expect(err).ToNot(HaveOccurred())
+				libwait.WaitForSuccessfulVMIStart(vmi)
+			})
+
+			It("should successfully start with hook sidecar annotation for v1alpha2", func() {
+				By("Starting a VMI")
+				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi)
+				vmi.ObjectMeta.Annotations = RenderValidQEMUArgsSidecar(hooksv1alpha2.Version)
+				Expect(err).ToNot(HaveOccurred())
+				libwait.WaitForSuccessfulVMIStart(vmi)
+			})
+
+			It("should call Collect and OnDefineDomain on the hook sidecar", func() {
+				By("Getting hook-sidecar logs")
+				vmi, err = virtClient.
+					VirtualMachineInstance(testsuite.GetTestNamespace(nil)).
+					Create(context.Background(), vmi)
+				Expect(err).ToNot(HaveOccurred())
+				logs := func() string { return getHookSidecarLogs(virtClient, vmi) }
+				libwait.WaitForSuccessfulVMIStart(vmi)
+				Eventually(logs,
+					11*time.Second,
+					500*time.Millisecond).
+					Should(ContainSubstring("Info method has been called"))
+				Eventually(logs,
+					11*time.Second,
+					500*time.Millisecond).
+					Should(ContainSubstring("OnDefineDomain method has been called"))
```

Comment:
nit: I think you can combine the two:

```suggestion
				Eventually(logs,
					11*time.Second,
					500*time.Millisecond).
					Should(And(ContainSubstring("Info method has been called"), ContainSubstring("OnDefineDomain method has been called")))
```

Reply:
Sure, I'll give it a try. Thanks!